packages/core/src/Site/TemplateConfig.ts

```diff
@@ -0,0 +1,61 @@
+import { SiteConfigPage } from './SiteConfig';
+
+type NjkFileVars = {
+  variableName: string,
+  fileSubstitutes: string[]
+};
+
+export type TemplateConfig = {
+  njkSubs: NjkFileVars[],
+  njkFile: string,
+  layoutObjs: SiteConfigPage[],
+  hasAutoSiteNav: boolean,
+  siteNavIgnore?: string[]
+};
+
+const defaultTemplate: TemplateConfig = {
+  njkSubs: [
+    {
+      variableName: 'siteNav',
+      fileSubstitutes: ['_Sidebar.md'],
+    },
+    {
+      variableName: 'footer',
+      fileSubstitutes: ['_Footer.md'],
+    },
+  ],
+  njkFile: 'defaultLayout.njk',
+  layoutObjs: [
+    { glob: '**/*.md', layout: 'default.md' },
+    { glob: '404.md', layout: '404.md' },
+  ],
+  hasAutoSiteNav: true,
+  siteNavIgnore: [
+    'index.md', '404.md',
+    'contents/topic1.md', 'contents/topic2.md', 'contents/topic3a.md', 'contents/topic3b.md',
```

Comment:
(not your pr but possible future work)

For a _conversion_ feature, what do you think of including `'contents/topic1.md', 'contents/topic2.md', 'contents/topic3a.md', 'contents/topic3b.md'`? (personally only just realised its a little odd we are adding pages when doing conversion)

`index.md` and `404.md` makes some sense but I don't think we have `file.exists` checks or fallback plans if the site already has these files too.